docs/source/python/timestamps.rst

```diff
@@ -0,0 +1,179 @@
+
+#.  Timezone information is lost.
+#.  Timestamps are truncated to microseconds.
+#.  The session time zone might have unintuitive impacts on 
+    translation of timestamp values. 
+
+Spark to Pandas (through Apache Arrow)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following cases assume the Spark configuration
+``spark.sql.execution.arrow.enabled`` is set to ``"true"``.
+
+::
+
+    >>> pdf = pd.DataFrame({'naive': [datetime(2019, 1, 1, 0)], 
+    ...                     'aware': [Timestamp(year=2019, month=1, day=1, 
+    ...                               nanosecond=500, tz=timezone(timedelta(hours=-8)))]})
+    >>> pdf
+           naive                               aware
+           0 2018-10-01 2018-10-01 00:00:00.000000500-08:00
+
+    >>> spark.conf.set("spark.sql.session.timeZone", "UTC")
+    >>> utc_df = sqlContext.createDataFrame(pdf).
+    >>> utf_df.show()
+    +-------------------+-------------------+
+    |              naive|              aware|
+    +-------------------+-------------------+
+    |2019-01-01 00:00:00|2019-01-01 08:00:00|
+    +-------------------+-------------------+
+                    
+Note that conversion of the aware timestamp is shifted to reflect the time
+assuming UTC (it represents the same instant in time).  For naive
+timestamps, Spark treats them as being in the system local
+time zone and converts them UTC.
+
+Now if the session time zone is set to US Pacific Time (PST) we don't
+see any shift in the display of the aware time zone (it
+still represents the same instant in time):
+
+::
+
+    >>> spark.conf.set("spark.sql.session.timeZone", "US/Pacific")
+    >>> pst_df = sqlContext.createDataFrame(pdf)
+    >>> pst_df.show()
+    +-------------------+-------------------+
+    |              naive|              aware|
+    +-------------------+-------------------+
+    |2019-01-01 00:00:00|2019-01-01 00:00:00|
+    +-------------------+-------------------+
+
+Looking again at utc_df.show() we see one of the impacts of the session time
+zone.  The naive timestamp was initially converted assuming UTC, the instant it
+reflects is actually earlier than the naive time zone from the PST converted
+data frame:
+
+::
+
+    >>> utc_df.show()
+    +-------------------+-------------------+
+    |              naive|              aware|
+    +-------------------+-------------------+
+    |2018-12-31 16:00:00|2019-01-01 00:00:00|
+    +-------------------+-------------------+
+
+Spark to Pandas
+~~~~~~~~~~~~~~~
+
+We can observe what happens when converting back to Arrow/Pandas.  Assuming the
+session time zone is still PST:
+
+::
+
+   >>> pst_df.show()
+   +-------------------+-------------------+
+   |              naive|              aware|
+   +-------------------+-------------------+
+   |2019-01-01 00:00:00|2019-01-01 00:00:00|
+   +-------------------+-------------------+
+
+   
+    >>> pst_df.toPandas()
+    naive      aware
+    0 2019-01-01 2019-01-01
```

Comment:
Can you show what Spark SQL schema has been assigned? Is the type of "aware" DATETIME WITH TIME ZONE or DATETIME WITHOUT TIME ZONE?